### DISCUSSION TITLE
Vertex shaders that manipulate vertex positions?

### DISCUSSION THREAD

**Discussion by redstonedash** [google/filament#4969] [Q&A] [answered]
I was reading through the material documentation about vertex shaders, and it only covers modifying material properties of vertices. Is their a way to change the positions?

**Comment by romainguy** [comment, marked as answer]
You can modify the worldPosition property for this.

**Reply by redstonedash** [reply]
Thank you. The name through me off.

**Reply by redstonedash** [reply]
Do you think you could provide a simple example of a vertex shader? can't get mine to compile.
(Edit: maybe that should have been my initial question, not sure if anyone will see this without it being reposted as a new question)
(Edit2: the world position is a vec4 not a vec3 be careful)

**Reply by bejado** [reply]
Our heightfield sample uses a [custom vertex shader](https://github.com/google/filament/blob/main/samples/materials/heightfield.mat#L20).
